### PR TITLE
Reorganize Concepts section (and others)

### PR DESCRIPTION
**Pull request by rosiel**
## Purpose / why

I didn't think that the "Concepts" section made much sense with what it had in it. I rewrote the section with some new pages outlining what I think are the key islandora concepts:
- nodes in islandora 
- media in islandora
- derivatives
- RDF

Then, I reorganized some of the files, putting more walkthroughs into the "Tutorials" section (which I renamed "walkthroughs"), and divvied the "Sysadmin documentation" into developer and user and contributing documentation.

I also created a section to discuss things like "collections" and "paged content" and hopefully "audio" "video" "image" and specifically "newspaper" - all the content types that have associated behaviours that people are going to want to see.

## Verification

Does having the "concepts" like this make sense?

## Interested Parties
* @islandora-community 
* @Islandora/documentation
* @Islandora/committers

### Pull-request Reviewer
Pull-request reviewer should ensure the following:

* [ ] Does this PR link to related [issues](https://github.com/Islandora/documentation/issues/)?
* [ ] Does the proposed documentation align with the [Islandora Documentation Style Guide](https://islandora.github.io/documentation/contributing/docs_style_guide/)?
* [ ] Are the changes accurate, useful, free of typos, etc?

### Person Merging
The person merging should ensure the following:

* [ ] Does mkdocs still build successfully? (This is indicated by TravisCI passing. To test locally, and see warnings, see [How To Build Documentation](https://islandora.github.io/documentation/technical-documentation/docs-build/).)
* [ ] If pages are renamed or removed, have all internal links to those pages been fixed?
* [ ] If pages are added, have they been linked to or placed in the menu?
* [ ] Did the PR receive at least one approval from a committer, and all issues raised have been addressed?

### CODE REVIEW

File: docs/concepts/rdf.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Islandora was build on the idea that a repository can be represented
+in Drupal in a way that can be mapped to RDF and present the repository 
+as Linked Data.
+
+## Portland Common Data Model
+
+Islandora uses the [Portland Common Data Model (PCDM)](https://github.com/duraspace/pcdm/wiki)
+ to arrange elements
+of the repository. Nodes correspond to pcdm:Objects, and Media to pcdm:Files.
+The "Member of" field on nodes allows us to create pcdm:memberOf relationships
```
Comment (islandora-community) [inline docs/concepts/rdf.md:12]: Checking - should this stay "Member of" or be referred to as `field_member_of` here?

**Review by islandora-community — APPROVED** [review]
This PR makes important changes to improve navigation and content of concepts documentation. The derivatives diagram/lane representation is a great addition!

File: docs/concepts/rdf.md (hunk 1 above)
Comment (rosiel) [reply docs/concepts/rdf.md:12]: I'm pretty happy leaving this in "user language" in this place.

File: docs/concepts/rdf.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# RDF in an Islandora Repository
+
+Islandora was build on the idea that a repository can be represented
```
Comment (ysuarez) [inline docs/concepts/rdf.md:3]: I think we may need to replace "build" with "built" here?

**Comment by ysuarez** [comment]
I need to build the docs based on this PR to see the new proposed TOC order side by side with the current one, to help we weigh in on the new order better. So far I like what I see.

**Comment by islandora-community** [comment]
Potential overlap with some content on the Resource Nodes and Media pages https://islandora.github.io/documentation/user-documentation/resource-nodes/ @inikolaidis

File: mkdocs.yml [hunk 3]
```diff
@@ -50,20 +50,19 @@ nav:
...
-          # Conceptual/ reference, all user roles/ sys admins/ developers:
-          # procedural information should be moved out installation guides
-      - 'Modelling content in Islandora  vs. Legacy': 'user-documentation/objects_to_resource_nodes.md'
-              # conceptual, all user roles
-              # moved from User documentation > Content in Islandora
-      - Islandora Defaults: 'reference/islandora_defaults_reference.md'
+      - 'Paged Content': 'user-documentation/paged-content.md'
+              # conceptual and procedural, repository managers/ curators
+              # move out procedural description for ordering pages
+      - 'Audio': 'models/audio.md'
```
Comment (islandora-community) [inline mkdocs.yml:62]: These 2 pages on models (audio, video) are to be made still, correct? I don't think we have a corresponding page in our documentation for these yet

**Comment by kstapelfeldt** [comment]
Thanks @rosiel for your awesome work.

**Pull request merged**